cranelift/isle/isle/src/sema.rs

```diff
@@ -957,23 +957,16 @@ impl TypeEnv {
 
         // Now collect types for extern constants.
         for def in &defs.defs {
-            match def {
-                &ast::Def::Extern(ast::Extern::Const {
-                    ref name,
-                    ref ty,
-                    pos,
-                }) => {
-                    let ty = match tyenv.get_type_by_name(ty) {
-                        Some(ty) => ty,
-                        None => {
-                            tyenv.report_error(pos, "Unknown type for constant");
-                            continue;
-                        }
-                    };
-                    let name = tyenv.intern_mut(name);
-                    tyenv.const_types.insert(name, ty);
-                }
-                _ => {}
+            if let ast::Def::Extern(ast::Extern::Const { name, ty, pos }) = def {
```

Comment:
Converting `match` to `if` helps a lot, but let's put the `&`/`ref` back in this pattern.